src/metabase/pulse/filters.clj

```diff
@@ -0,0 +1,40 @@
+(ns metabase.pulse.filters
+  "Utilities for processing dashboard filters for inclusion in dashboard subscriptions."
+  (:require [clojure.string :as str]
+            [metabase.config :as config]
+            [metabase.util :as u]
+            [metabase.util.urls :as url]))
+
+(defn merge-filters
+  "Returns a definitive list of filters applied to a dashboard subscription, combining the :parameters field on the
+  subscription with any filters with default values set on the dashboard.
+
+  Only default dashboard subscriptions are allowed for non-EE builds."
+  [subscription dashboard]
+  (let [subscription-filters       (if config/ee-available?
```

Comment:
Is this an OK pattern to use to determine whether to include EE-only logic? Or do we need to actually define two separate versions of this function in different parts of the codebase?

Reply:
I think you'll want to use this https://github.com/metabase/metabase/blob/552ab7d94a79af42da6f376c2f6e283929bdd79e/src/metabase/pulse.clj#L31-L36

`ee-available?` just checks for the presence of ee code but not whether we have a valid token or not.

And I think the separation is about license enforcement as well. If we just put an `if` here, then it would be completely legal for someone to come and remove this `if` as the OSS code is AGPL.

Reply:
Good call, I didn't realize that code already existed.